src/utils/map-controls.js

```diff
@@ -121,16 +121,19 @@ export default class MapControls {
    * Extract interactivity options
    */
   setOptions({
+    // TODO(deprecate): remove this when `onChangeViewport` gets deprecated
```

Comment:
Same here - InteractiveMap delegate the handling of `onChangeViewport`, `dragRotate`, `scrollZoom` etc. to this class. Technically these props are not owned by InteractiveMap, and may be ignored if a custom controller is supplied.